tests/test_job_utils.py

```diff
@@ -69,6 +71,48 @@ def test_job_serialisation_deserialisation(command_string, expected_output):
     ).decode("utf-8")
 
 
+@pytest.mark.parametrize(
+    "cpu,output",
+    [
+        ("100m", True),
+        ("250m", True),
+        ("1", True),
+        ("2.5", True),
+        ("0.1", True),
+        ("0.001", True),
+        ("1500m", True),
+        ("0m", False),
+        ("-100m", False),
+        ("1.5m", False),
+        ("2 cores", False),
+        ("one", False),
+        ("1000millicores", False),
+    ],
+)
+def test_validate_kubernetes_cpu_format(cpu, output):
+    """Test validation of K8s CPU format."""
+    assert validate_kubernetes_cpu(cpu) is output
+
+
+@pytest.mark.parametrize(
+    "k8s_cpu,millicores",
+    [
+        (100, 100000),
+        (0.1, 100),
+        (1, 1000),
+        (1.5, 1500),
+        ("100m", 100),
+        ("250m", 250),
+        ("0.5", 500),
+        ("2.25", 2250),
+        ("1500m", 1500),
+    ],
+)
+def test_kubernetes_cpu_to_millicores(k8s_cpu, millicores):
+    """Test conversion of k8s CPU format to millicores."""
```

Comment:
Cosmetics; In textual documentation, the usual abbreviation is "K8s" i.e. uppercase K.